src/jdk.management/share/classes/com/sun/management/HotSpotDiagnosticMXBean.java

```diff
@@ -117,4 +118,53 @@ public interface HotSpotDiagnosticMXBean extends PlatformManagedObject {
      *     ManagementPermission("control").
      */
     public void setVMOption(String name, String value);
+
+    /**
+     * Generate a thread dump to the given file in the given format. The file
+     * must not exist. The {@code outputFile} parameter must be an absolute path.
+     *
+     * <p> The thread dump will include output for all platform threads. It may
+     * include output for some or all virtual threads.
+     *
+     * @implSpec
+     * The default implementation throws {@code UnsupportedOperationException}.
+     *
+     * @apiNote
+     * The output file is required to be an absolute path as the MXBean may be
+     * accessed remotely from a tool or program with a different current user
+     * directory.
+     *
+     * @param  outputFile the path to the file to create
+     * @param  format the format to use (TEXT_PLAIN or JSON)
```

Comment:
I think it might be better if the description is changed to "the format to use". There is already a link to the ThreadDumpFormat enum in method signature.